daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -521,6 +522,7 @@ private[lf] class Runner(
                     )
                   case cmd: ScriptF.Cmd =>
                     cmd.execute(env).transform {
+                      case f @ Failure(Runner.Canceled) => f
```

Comment:
@akrmn your test actually catch this issue: the `Runner.Canceled` form the `IdeLedger` was swallowed.

Reply:
aah I see it now, thank you Rémy!